Source/AviationWeather.Test/Source/metar_parser_tests.cpp

```diff
@@ -0,0 +1,575 @@
+    TEST_METHOD(METAR_Parser_Altimeter);
+    TEST_METHOD(METAR_Parser_Remarks);
+};
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_ReportType()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(metar::report_type::metar, metar.type);
+
+    // METAR
+    std::string m1("METAR ");
+    metar::parsers::parse_report_type(metar, m1);
+    Assert::AreEqual(metar::report_type::metar, metar.type);
+
+    // SPECI
+    std::string m2("SPECI ");
+    metar::parsers::parse_report_type(metar, m2);
+    Assert::AreEqual(metar::report_type::special, metar.type);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_StationIdentifier()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(std::string(""), metar.station_identifier);
+
+    // KSFO
+    std::string m1("KSFO ");
+    metar::parsers::parse_station_identifier(metar, m1);
+    Assert::AreEqual(std::string("KSFO"), metar.station_identifier);
+
+    // EBGE
+    std::string m2("EGBE ");
+    metar::parsers::parse_station_identifier(metar, m2);
+    Assert::AreEqual(std::string("EGBE"), metar.station_identifier);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_ObservationTime()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(uint8_t(1), metar.report_time.day_of_month);
+    Assert::AreEqual(uint8_t(0), metar.report_time.hour_of_day);
+    Assert::AreEqual(uint8_t(0), metar.report_time.minute_of_hour);
+
+    // 8th, 17:53
+    std::string m1("081753Z ");
+    metar::parsers::parse_observation_time(metar, m1);
+    Assert::AreEqual(uint8_t(8), metar.report_time.day_of_month);
+    Assert::AreEqual(uint8_t(17), metar.report_time.hour_of_day);
+    Assert::AreEqual(uint8_t(53), metar.report_time.minute_of_hour);
+
+    // 20th, 03:28
+    std::string m2("200328Z ");
+    metar::parsers::parse_observation_time(metar, m2);
+    Assert::AreEqual(uint8_t(20), metar.report_time.day_of_month);
+    Assert::AreEqual(uint8_t(3), metar.report_time.hour_of_day);
+    Assert::AreEqual(uint8_t(28), metar.report_time.minute_of_hour);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_Modifier()
+{
+    metar::metar_info metar("");
+    Assert::AreEqual(metar::modifier_type::automatic, metar.modifier);
+
+    // AUTO
+    std::string m1("AUTO ");
+    metar::parsers::parse_modifier(metar, m1);
+    Assert::AreEqual(metar::modifier_type::automatic, metar.modifier);
+
+    // COR
+    std::string m2("COR ");
+    metar::parsers::parse_modifier(metar, m2);
+    Assert::AreEqual(metar::modifier_type::corrected, metar.modifier);
+}
+
+//-----------------------------------------------------------------------------
+
+void MetarParserTests::METAR_Parser_Wind()
```

Comment:
Can we add an additional test for winds greater than 100 knots?  AC 00-45G gives these examples for wind groups:

Notation | Meaning
-------- | -------
05008KT | Wind 50 degrees at 8 knots
15014KT | Wind 150 degrees at 14 knots
340112KT | Wind 340 degrees at 112 knots